source/manual/alerts/signon-api-user-token-expires-soon.html.md

```diff
@@ -1,63 +1,86 @@
+
+1. Choose **Add application token**.
+
+1. From the **Application** dropdown, select the application that matches the `application` field in the alert.
+
+1. Choose **Create access token**.
+
+The nightly `signon-sync-token-secrets-to-k8s` cronjob will update the token in the Kubernetes secret that the application uses.
+
+### 2. Check that the client application works with the new token
+
+1. Trigger the token sync cronjob to run now (or wait until the next day).
 
-> Check the *Last synced at* time to see if the API User is still using the
-> application. If you are confident the token is unused, the you can just
-> click the *Revoke* button to remove it and there's no need to continue.
+    ```sh
+    k create job --from cronjob/signon-sync-token-secrets-to-k8s signon-token-sync-$USER
+    ```
 
-- Click *Add application token*, select the application and click *Create
-  access token*.
+   The job should finish within a few seconds. You can view the job's output to check that it succeeded:
 
-- At this point the new token is created, and will be picked up automatically by a job that runs once a day (around 1am) to sync tokens to the applications in Kubernetes. If you want to check that the token has been updated, you can wait until the next day and query the app: `kubectl -n apps exec -it deploy/<your-app-name> -- printenv | grep BEARER_TOKEN`, then find the relevant token in the output and compare the first and last 8 characters with the hidden token shown in Signon.
+    ```sh
+    k logs -f job/signon-token-sync-$USER
+    ```
 
-## Updating the tokens immediately
+1. Open a Rails console on the client app. For example, if the `api_user` is `short-url-manager@alphagov.co.uk`, open a console on `short-url-manager`.
 
-If the tokens have actually expired, or need to be rotated immediately in an emergency, you won't want to wait for the daily sync. You can request it to run immediately:
+    ```sh
+    k exec -it deploy/short-url-manager -- rails c
+    ```
 
-- `kubectl create job --from cronjob/signon-sync-token-secrets-to-k8s signon-token-sync-$USER`
-- You will then need to restart the relevant app to pick up the new token: `k rollout restart deploy/<your-app-name>`
+1. Fetch the environment variable that contains the token. The variable name refers to the server (destination) application, which is the same as the `application` field in the alert. For example, for short-url-manager (client) talking to publishing-api (server):
 
-At this point you can check that the app has the new token as above.
+    ```ruby
+    ENV["PUBLISHING_API_BEARER_TOKEN"]
+    ```
 
-## Revoke the old token
+    Check that the result matches the new token in Signon and not the old one.
 
-Once the new token is in place, you should check that it can be used:
+1. You can also check that the token works by making an API call yourself from the client app's Rails console. For example, to check that Short URL Manager can talk to Publishing API:
```

Comment:
```suggestion
1. You can also check that the token works by making an API call yourself from the client app's Rails console. For example, to check that Short URL Manager can talk to Publishing API ([as per the API's documentation](https://github.com/alphagov/publishing-api/blob/main/docs/api.md)):
```
The other thing that we thought might be useful to include was a link to the API documentation, so it's clearer what to call for other APIs.

Reply:
Linked to gds-api-adaptors since the Rubydoc in there is pretty good and corresponds directly to the method calls that you'd need to make.